package.json

```diff
@@ -68,14 +71,22 @@
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.4.3",
     "eslint-plugin-testing-library": "^6.2.0",
+    "jsdom": "24.0.0",
+    "lint-staged": "^15.2.8",
     "prettier": "^3.1.1",
     "tsc-files": "^1.1.4",
     "typescript": "^5.3.3",
     "vite": "^5.3.4",
     "vite-tsconfig-paths": "^4.2.3",
-    "vitest": "^1.1.3",
-    "@vitest/coverage-v8": "1.2.1",
-    "@vitest/ui": "1.2.1",
-    "jsdom": "24.0.0"
+    "vitest": "^1.1.3"
+  },
+  "resolutions": {
+    "@graphql-codegen/cli": "^5.0.0",
+    "graphql": "^16.8.1"
+  },
+  "lint-staged": {
+    "*": "pnpm graphql-gen",
```

Comment:
I assume the `"*"` indicates the file extension, you can make it `"*.{gql,graphql}"` because it should only react if these filetypes are staged.

Reply:
@pylipp @HaGuesto  Quick note on that: you might need to have the local frontend toolchain if you want to generate these files on commit.